### DISCUSSION TITLE
Question: Modifying Microsoft Basic Solution to Use Timer

### DISCUSSION THREAD

**Discussion by rzuckerm** [PlummersSoftwareLLC/Primes#941] [Q&A] [answered]
I recently discovered that Microsoft Basic (really Commodore 64 Basic) has a variable called `TI` that indicates how long the current program has been executing in 1/60th of second resolution. Currently, `PrimeBasic/solution_4` uses a bash script to do the actual timing and results output, and there are separate Basic programs that do solution validation. I would like to modify the bash script so that all it does is run the boolean and bit versions, and let the Basic programs do the timing, validation, and results output. I did a prototype on the boolean version, and it seems to work (although it is slower since it needs to reinitialize the array for every pass). Before I do an actual PR with this change, I'd like to know if this is an acceptable change to make.

**Comment by rbergen** [comment, marked as answer]
I think I may have caused some confusion by the selection of `cbmbasic` as the emulator for running the Microsoft BASIC v2 solution (`PrimeBASIC/solution_4`) in the context of this project.  

The solution in question was actually added from the perspective of the Altair 8800, in the sense that that was the machine it was written for and originally tested on. As far as I know, Microsoft BASIC on the Altair did not include the Jiffy timer based TI variables that are available on Commodore machines, as the vast majority of machines on which Microsoft BASIC v2 was available also didn't. This is because Commodore BASIC was indeed a derivation of Microsoft BASIC, but with Commodore-specific extensions.

Considering the origin and the purpose of the solution (i.e. offering an implementation that would run on "all" Microsoft BASIC v2 versions that were "out in the field" back in the day) I think we need to limit ourselves to the least common denominator. In this case, that means we should assume TI is not available.

For completeness, I have to note that the current `PrimeBASIC/solution_4` already no longer runs on the first, 4K version of Microsoft BASIC on the Altair 8800, because it uses some features only available on later versions of Microsoft BASIC that use more memory. An example of such a feature is it using variable names that consist of multiple letters; the very first version of Microsoft BASIC only supported variable names consisting of one letter followed by a maximum of one digit. As the solution would not run on a 4K Altair anyway because it needs (far) more memory to calculate the sieve up to 10,000, I think that is acceptable.
